### PR TITLE
Added password strength validation on signup page

### LINKED ISSUE #25: [Feature Request]: Adding password validation on signup page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Password validation is crucial for ensuring user accounts are secure. Weak passwords can be easily compromised, leading to unauthorized access and potential data breaches.

Password strength is a measure of how resistant a password is to being guessed or cracked. Strong passwords typically:

Are at least 8-12 characters long.
Include a mix of upper and lower case letters.
Contain numbers and special characters.
Avoid common words and easily guessable patterns.

This can be added with proper UI for representing strength based on colour like red, orange, green

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue
- [X] I'm willing to provide further clarification or assistance if needed.

### PR DESCRIPTION
**Pull request by jvkousthub**
- Fixes #25 

## 📋 Description

Turns into different colours depending on the password strength

## 🔨 Changes Made

- The input changes to different color based on the password strength
- weak password returns red
- medium gives orange
- strong password gives green
- solves #25 

## 🔨 Screen Shots OR Short videos of working

image
image
image

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] Read Contributing.md

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [x] New feature (non-breaking change which adds functionality) ✨
- [x] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to Talk-Trove! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

**Comment by jvkousthub** [comment]
@kartikayasija Please review and let me know if any changes required

File: frontend/src/components/Authentication/Signup.js [hunk 1]
```diff
@@ -20,6 +20,44 @@ const Signup = () => {
...
+    else{
+      return "medium";
+    }
+  }
+  const handlePasswordChange = (e)=>{
+    const newPassword = e.target.value;
+    setPassword(newPassword);
+    setPasswordStrength(checkPasswordStrength(newPassword));
+  }
+  const getPasswordStrengthColor=()=>{
```
Comment (Sibam-Paul) [inline frontend/src/components/Authentication/Signup.js:49]: U have added color make it like the user can't signup without giving strong password and give a hidden response when the user enters low strength password
like 
image

like here

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

**Comment by jvkousthub** [comment]
@Sibam-Paul please check

**Comment by Sibam-Paul** [comment]
Add a video plz

**Pull request closed without merging**